.circleci/config.yml

```diff
@@ -0,0 +1,205 @@
+# For a detailed guide to building and testing on iOS, read the docs:
+# https://circleci.com/docs/2.0/testing-ios/
+
+version: 2.1
+
+# Workflows orchestrate a set of jobs to be run;
+workflows:
+  version: 2
+  build-test-deploy:
+    jobs:
+      - build-and-test
+orbs:
+  slack: circleci/slack@4.0.2
+
+jobs:
+
+  build-and-test:
+    macos:
+      xcode: 12.0.0 # Specify the Xcode version to use
+
+    steps:
+      - checkout
+
+      # restore pods related caches
+      - restore_cache:
+          keys:
+            - 1-gems-{{ checksum "Gemfile.lock" }}
+
+      # make sure we're on the right version of cocoapods
+      - run:
+          name: Verify Cocoapods Version
+          command: bundle check || bundle install --path vendor/bundle
+
+      # save cocoapods version gem data
+      - save_cache:
+          key: 1-gems-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+
+      # restore pods related caches
+      - restore_cache:
+          keys:
+            - cocoapods-cache-v5-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
+            - cocoapods-cache-v5-{{ arch }}-{{ .Branch }}
+            - cocoapods-cache-v5
+
+      # install CocoaPods - using default CocoaPods version, not the bundle
+      - run:
+          name: Repo Update & Install CocoaPods
+          command: make ci-pod-install
+
+      # save pods related files
+      - save_cache:
+          name: Saving CocoaPods Cache
+          key: cocoapods-cache-v4-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
+          paths:
+            - ./Pods
+            - ~/.cocoapods
+
+      - run:
+          name: Lint Source Code
+          command: make lint
+
+      - run:
+          name: Install nicklockwood/SwiftFormat
+          command: make install-swiftformat
+
+      - run:
+          name: Check Source Code formatting
+          command: make check-format
+
+      # pre-start the simulator to prevent timeouts
+      - run:
+          name: Pre-start Simulator
+          command: xcrun instruments -w "iPhone 8 (13.1) [" || true
+
+      - run:
+          name: Run Tests
+          command: make test
+
+      # code coverage
+      - run:
+          name: Upload Code Coverage Report
+          command: |
+             curl -s https://codecov.io/bash > codecov;
+             VERSION=$(grep 'VERSION=\"[0-9\.]*\"' codecov | cut -d'"' -f2);
+             for i in 1 256 512
+             do
+               shasum -a $i -c --ignore-missing <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM") ||
+               shasum -a $i -c <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM" | grep -w "codecov")
+             done
+             bash ./codecov -v -X s3 -c -D "./out/Build/ProfileData" -J "AEPCampaignClassic" -t ${CODECOV_TOKEN}
```

Comment:
probably a good idea.  also would be good to have all the dev work going to a `dev` branch :)

Reply:
I remember we had limitations in the capacity to use circleCI on private repo's.
Wondering we can still keep them public since this will be released in next two weeks

+1 @sbenedicadb I will move any further development to dev branch